bundle/ScheduledVisibility/Handler/Location.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Netgen\Bundle\IbexaScheduledVisibilityBundle\ScheduledVisibility\Handler;
+
+use Ibexa\Contracts\Core\Repository\LocationService;
+use Ibexa\Contracts\Core\Repository\Repository;
+use Ibexa\Contracts\Core\Repository\Values\Content\Content;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\ScheduledVisibility\ScheduledVisibilityInterface;
+
+final class Location implements ScheduledVisibilityInterface
+{
+    public function __construct(
+        private readonly Repository $repository,
+        private readonly LocationService $locationService,
+    ) {}
+
+    public function hide(Content $content): void
+    {
+        $locations = $this->repository->sudo(
+            fn () => $this->locationService->loadLocations($content->getContentInfo()),
+        );
+
+        foreach ($locations as $location) {
+            $this->repository->sudo(
+                fn () => $this->locationService->hideLocation($location),
+            );
+        }
+    }
+
+    public function reveal(Content $content): void
+    {
+        $locations = $this->repository->sudo(
+            fn () => $this->locationService->loadLocations($content->getContentInfo()),
+        );
+
+        foreach ($locations as $location) {
+            $this->repository->sudo(
+                fn () => $this->locationService->unhideLocation($location),
+            );
+        }
+    }
+
+    public function isHidden(Content $content): bool
+    {
+        return !$this->isVisible($content);
+    }
+
+    public function isVisible(Content $content): bool
+    {
+        $locations = $this->locationService->loadLocations($content->contentInfo);
```

Comment:
Should also use `sudo()` as we might not have permissions to load all Locations.

Reply:
fixed in 091f0e24f3b9c40dd141508041b8403dee4c8b21